src/EditorFeatures/CSharp/AutomaticCompletion/Sessions/CurlyBraceCompletionSession.cs

```diff
@@ -193,27 +193,45 @@ private int GetDesiredIndentation(IBraceCompletionSession session, ITextSnapshot
             return openingPoint - openingSpanLine.Start;
         }
 
-        private class BraceCompletionFormattingRule : BaseFormattingRule
+        private sealed class BraceCompletionFormattingRule : BaseFormattingRule
         {
             private static readonly Predicate<SuppressOperation> s_predicate = o => o == null || o.Option.IsOn(SuppressOption.NoWrapping);
 
             private static readonly ImmutableArray<BraceCompletionFormattingRule> s_instances = ImmutableArray.Create(
-                new BraceCompletionFormattingRule(FormattingOptions.IndentStyle.None),
-                new BraceCompletionFormattingRule(FormattingOptions.IndentStyle.Block),
-                new BraceCompletionFormattingRule(FormattingOptions.IndentStyle.Smart));
+                new BraceCompletionFormattingRule(FormattingOptions.IndentStyle.None, CSharpFormattingOptions.NewLinesForBracesInObjectCollectionArrayInitializers.DefaultValue),
+                new BraceCompletionFormattingRule(FormattingOptions.IndentStyle.Block, CSharpFormattingOptions.NewLinesForBracesInObjectCollectionArrayInitializers.DefaultValue),
+                new BraceCompletionFormattingRule(FormattingOptions.IndentStyle.Smart, CSharpFormattingOptions.NewLinesForBracesInObjectCollectionArrayInitializers.DefaultValue));
 
             private readonly FormattingOptions.IndentStyle _indentStyle;
+            private readonly bool _newLinesForBracesInObjectCollectionArrayInitializers;
 
-            private BraceCompletionFormattingRule(FormattingOptions.IndentStyle indentStyle)
-                => _indentStyle = indentStyle;
+            private BraceCompletionFormattingRule(FormattingOptions.IndentStyle indentStyle, bool newLinesForBracesInObjectCollectionArrayInitializers)
+            {
+                _indentStyle = indentStyle;
+                _newLinesForBracesInObjectCollectionArrayInitializers = newLinesForBracesInObjectCollectionArrayInitializers;
+            }
 
             public static AbstractFormattingRule ForIndentStyle(FormattingOptions.IndentStyle indentStyle)
             {
                 Debug.Assert(s_instances[(int)indentStyle]._indentStyle == indentStyle);
                 return s_instances[(int)indentStyle];
             }
 
-            public override AdjustNewLinesOperation GetAdjustNewLinesOperation(SyntaxToken previousToken, SyntaxToken currentToken, AnalyzerConfigOptions options, in NextGetAdjustNewLinesOperation nextOperation)
+            public override AbstractFormattingRule WithOptions(AnalyzerConfigOptions options)
+            {
+                var newLinesForBracesInObjectCollectionArrayInitializers = options.GetOption(CSharpFormattingOptions.NewLinesForBracesInObjectCollectionArrayInitializers);
+
+                if (newLinesForBracesInObjectCollectionArrayInitializers == _newLinesForBracesInObjectCollectionArrayInitializers)
```

Comment:
1. cute :)   maybe add explanatory comment on the field?
2. how many values can this option have? It's a bool right?  for people not on the default, can we cache that as well?  so just have two cached rules and only return one or the other.

Reply:
This was the old pattern. I meant to convert them all to structs.

Reply:
Fixed now